spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -758,20 +789,99 @@ def write_sorting(sorting, save_path, **nwbfile_kwargs):
             #         waveform_sd=traces_std
             #     )
 
-            # Units spike features
-            nspikes = {k: get_nspikes(nwbfile.units, int(k)) for k in ids}
+            nspikes = {k: get_nspikes(nwbfile.units, int(k)) for k in unit_ids}
             for ft in all_features:
-                vals = [v[ft] if ft in v else [np.nan] * nspikes[int(k)]
-                        for k, v in sorting._features.items()]
-                flatten_vals = [item for sublist in vals for item in sublist]
-                nspks_list = [sp for sp in nspikes.values()]
-                spikes_index = np.cumsum(nspks_list).tolist()
-                set_dynamic_table_property(
-                    dynamic_table=nwbfile.units,
-                    row_ids=ids,
-                    property_name=ft,
-                    values=flatten_vals,
-                    index=spikes_index,
-                )
+                values = []
+                skip_feature = False
+                if not ft.endswith('_idxs'):
```

Comment:
Refractored the features part a bit. A commn case is to compute features only on a subset of spikes. In that case, SI creates a feature called e.g. `waveforms` and one called `waveforms_idxs`, containing the indexes. Of course the latter case was causing errors, so for let's skip those features in favor of *full* features.